### PR TITLE
Parse imports correctly (built compilers)

### LINKED ISSUE #64: Fix imports parsing bug for docker / native
Overlooked this in the initial work. Solcjs has [a feature](https://github.com/ethereum/solc-js#from-version-021) we use to get import paths that's missing from the command-line compilers. Believe the simplest solution is to continue to use solcjs for this, but we'll likely need to fetch a js solc that matches the selected native version. 

*Think* this import path collection is fast (there's a note to that effect in the code). It's not a real compilation? Need to check that out too.

### PR DESCRIPTION
**Pull request by cgewecke**
Fixes #64. 

Uses solcjs to resolve import paths when running a built compiler. Solcjs needs to be matched to the built version, so the first time a built compiler runs there's additional overhead for the remote fetch (~5 sec). That said running with `docker` is more than twice as fast. Native might be faster. Zeppelin's entire suite (dozens of contracts) compiles in ~11 seconds vs. ~27 seconds (JS) on my box. Import parsing overhead without the fetch seems negligible, ~1 sec. 

Also:
+ Consolidated `get` built compilers into a single method
+ Re-organized test sources and added a test for this.

### CODE REVIEW

File: compilerProvider.js [hunk 1]
```diff
@@ -225,41 +225,36 @@ CompilerProvider.prototype.getByUrl = function(version){
...
-
-  return {
-    compileStandard: (options) => {
-      const command = 'docker run -i ethereum/solc:' + version + ' --standard-json';
-      const result = child.execSync(command, {input: options});
-      return String(result);
-    },
-
-    version: () => versionString,
+CompilerProvider.prototype.getBuilt = function(buildType){
```
Comment (gnidan) [inline compilerProvider.js:232]: Don't you tell me to `getBuilt`, CompilerProvider. Rude.

**Review by gnidan — APPROVED** [review]
This looks good.

It's weird that `solcjs` is necessary for the import parsing. It'd be nice if there were a way around this.

Just out of curiosity, what part fails when doing `parseImports` with the native solc?

**Comment by cgewecke** [comment]
The whole `parseImports` mechanism is built around a special JS callback that's documented [here](https://github.com/ethereum/solc-js#from-version-021). Not sure what we'd do instead because I don't see a comparable parsing option in the [command line solc docs](https://solidity.readthedocs.io/en/v0.4.23/using-the-compiler.html). We have quite a bit of pre-processing that doesn't necessarily fit into the way command-line solc is designed? Idk.

**Comment by gnidan** [comment]
> The whole `parseImports` mechanism is built around a special JS callback that's documented [here](https://github.com/ethereum/solc-js#from-version-021). Not sure what we'd do instead because I don't see a comparable parsing option in the [command line solc docs](https://solidity.readthedocs.io/en/v0.4.23/using-the-compiler.html). We have quite a bit of pre-processing that doesn't necessarily fit into the way command-line solc is designed? Idk. 

That's fascinating! Thanks for filling in the gaps in my mental model of this stuff.

Doing a bit of investigating for my own curiosity, [this comment](https://github.com/trufflesuite/truffle-compile/blob/develop/parser.js#L137-L139) seems to suggest we'd be able to avoid using that callback, given a bit of extra trouble.

Trying this out with the native `solc`, it looks like we could search for import errors ourselves - see [gist](https://gist.github.com/gnidan/c125abe66db0167c204cdcb732810667).

Not saying it's worth the effort, but it might be later.

**Comment by cgewecke** [comment]
@gnidan Yes agree this definitely seems possible. . .

**Comment by cgewecke** [comment]
Thought a bit more about this - probably worth profiling and might also require we revise the minimization algo to reduce computational complexity.

My impression while debugging for the PR was that solcjs parses this stuff very quickly ( ~ 1 sec for Zeppelin). `parseImports` runs for each file (potentially more than once in normal use). If solc is a `docker` this might be [bad](https://blog.iron.io/the-overhead-of-docker-run/) because spin-up could be in the 300ms - 900ms range depending on OS. We'd want to do that once, and possibly not at all. 

Would be nice if `solc` exposed the parser as a separate piece of functionality.

**Comment by gnidan** [comment]
@cgewecke ah, thanks for following up to find that out. Unfortunate that `solc` doesn't just expose the parser. I'll ask them if I should open an issue for that.

**Pull request merged**